client/src/component/FlightSearchComponent/FlightSearch.js

```diff
@@ -0,0 +1,146 @@
+import React, { useState } from 'react';
+
+import { Grid, Paper, TextField, Button, InputLabel } from '@material-ui/core';
+import { Autocomplete } from '@material-ui/lab';
+
+import useStyles from './styles';
+
+// Mock Data
+const cities = [
+  { title: 'Vancouver' },
+  { title: 'Calgary' },
+  { title: 'Toronto' },
+  { title: 'Bangkok' },
+];
+
+const FlightSearch = () => {
+  const classes = useStyles();
+
+  const curr = new Date();
+  curr.setDate(curr.getDate());
+  const date = curr.toISOString().substr(0, 10);
+
+  const [from, setFrom] = useState('Vancouver');
+  const [to, setTo] = useState('Bangkok');
+  const [arrival, setArrival] = useState(date);
+  const [departure, setDeparture] = useState(date);
+  const [travellers, setTravellers] = useState(1);
+
+  const onSelectFrom = (...[, value]) => {
+    setFrom(value);
+  };
+
+  const onSelectTo = (...[, value]) => {
+    setTo(value);
+  };
+
+  const handleArrival = (e) => {
+    const { value } = e.target;
+    setArrival(value);
+  };
+
+  const handleDeparture = (e) => {
+    const { value } = e.target;
+    setDeparture(value);
+  };
+
+  const handleTravellers = (e) => {
+    const { value } = e.target;
+    setTravellers(value);
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    const searchObject = {
+      from: from,
+      to: to,
+      arrival: arrival,
+      departure: departure,
+      travellers: travellers,
+    };
+  };
+
+  return (
+    <div className={classes.root}>
+      <form onSubmit={handleSubmit}>
+        <Paper className={classes.paperContainer} elevation={7}>
+          <Grid className={classes.input} lg={2} sm={3} xs={6} item>
+            <InputLabel>From</InputLabel>
+            <Autocomplete
+              freeSolo
+              id='from'
+              name='from'
+              options={cities.map((city) => city.title)}
+              defaultValue={from}
+              value={from}
+              onChange={onSelectFrom}
+              style={{ width: 150 }}
+              renderInput={(params) => <TextField name='from' {...params} />}
+            />
+          </Grid>
+          <Grid className={classes.input} lg={2} sm={3} xs={6} item>
+            <InputLabel>Where to go</InputLabel>
+            <Autocomplete
+              freeSolo
+              id='to'
+              name='to'
+              options={cities.map((city) => city.title)}
+              defaultValue={to}
+              value={to}
+              onChange={onSelectTo}
+              style={{ width: 150 }}
+              renderInput={(params) => <TextField name='to' {...params} />}
+            />
+          </Grid>
+          <Grid className={classes.input} lg={2} sm={3} xs={6} item>
+            <InputLabel>Arrival</InputLabel>
+            <TextField
+              id='date'
+              type='date'
+              name='departure'
+              defaultValue={arrival}
+              onChange={handleArrival}
```

Comment:
so the nice part about using the separate useState hooks is that now for each of these you can just do something like
`onChange={e => setArrival(e.target.value)}` 
and it'll be super clean 👍

Reply:
For some reason, I was told that it was bad practice to call the function in the actual onChange, hence why I did it the way I did. :P Just in case if you were wondering.